### PR TITLE
feature: code outline & breadcrumbs

### PR DESCRIPTION
**Pull request by zbenhadi**

### CODE REVIEW

File: pkg/ast/executor.go [hunk 1]
```diff
@@ -28,6 +30,7 @@ type BaseExecutor struct {
 	UserParameters      map[string]Parameter
 	UserParametersRange protocol.Range
 	Uncomplete          bool
+	Environment         Environment
```
Comment (rlegan) [inline pkg/ast/executor.go:33]: whats the difference between this and `Environment` in `ExecutableParameters` ?

File: pkg/parser/executors.go [hunk 2]
```diff
@@ -97,6 +97,7 @@ func (doc *YamlDocument) parseBaseExecutor(base *ast.BaseExecutor, nameNode *sit
 			base.BuiltInParameters.WorkingDirectory = doc.GetNodeText(valueNode)
 		case "environment":
 			base.BuiltInParameters.Environment = doc.parseDictionary(GetChildOfType(valueNode, "block_mapping"))
+			base.Environment = doc.parseEnvs(valueNode)
```
Comment (rlegan) [inline pkg/parser/executors.go:100]: we have two things that does the same thing ?

File: pkg/server/methods/textDocumentSymbols.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+func (methods *Methods) DocumentSymbols(reply jsonrpc2.Replier, req jsonrpc2.Request) error {
+	params := protocol.DocumentSymbolParams{}
+	if err := json.Unmarshal(req.Params(), &params); err != nil {
+		return reply(methods.Ctx, nil, fmt.Errorf("%s: %w", jsonrpc2.ErrParse, err))
+	}
+
+	res, err := languageservice.DocumentSymbols(params, methods.Cache, methods.LsContext)
+
+	// fmt.Println("Methods DocSymbols", res)
```
Comment (rlegan) [inline pkg/server/methods/textDocumentSymbols.go:20]: to delete

File: pkg/parser/jobs.go [hunk 4]
```diff
@@ -106,6 +106,14 @@ func (doc *YamlDocument) parseSingleJob(jobNode *sitter.Node) ast.Job {
 				machineNode = child
 				machineNodeFound = true
 
+				// TODO: Check this ish better. (will have to rework parseSingleExecMachine ...)
```
Comment (rlegan) [inline pkg/parser/jobs.go:109]: is it still necessasry ?

File: pkg/server/methods/textDocumentSymbols.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+	if err != nil {
+		return reply(
+			methods.Ctx,
+			nil,
+			err,
+		)
+	}
```
Comment (rlegan) [inline pkg/server/methods/textDocumentSymbols.go:28]: do we want to show an error in the client side or maybe show nothing ?

File: pkg/services/documentSymbols.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	"fmt"
+
+	yamlparser "github.com/CircleCI-Public/circleci-yaml-language-server/pkg/parser"
+	"github.com/CircleCI-Public/circleci-yaml-language-server/pkg/services/documentSymbols"
+	"github.com/CircleCI-Public/circleci-yaml-language-server/pkg/utils"
+	"go.lsp.dev/protocol"
+)
+
+func DocumentSymbols(params protocol.DocumentSymbolParams, cache *utils.Cache, context *utils.LsContext) ([]protocol.DocumentSymbol, error) {
+	fmt.Println("LanguageServer.DocumentSymbols")
```
Comment (rlegan) [inline pkg/services/documentSymbols.go:13]: to delete

File: pkg/server/methods/textDocumentSymbols.go (hunk 5 above)
Comment (zbenhadi) [reply pkg/server/methods/textDocumentSymbols.go:28]: I feel like it's spam I would rather not do it

File: pkg/parser/jobs.go (hunk 4 above)
Comment (zbenhadi) [reply pkg/parser/jobs.go:109]: Nope, fixed thank you

File: pkg/services/documentSymbols.go (hunk 6 above)
Comment (zbenhadi) [reply pkg/services/documentSymbols.go:13]: Removed

File: pkg/server/methods/textDocumentSymbols.go (hunk 3 above)
Comment (zbenhadi) [reply pkg/server/methods/textDocumentSymbols.go:20]: Deleted

File: pkg/services/documentSymbols/workflows.go [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+			Name:           "Jobs",
+			Range:          workflow.JobsRange,
+			SelectionRange: workflow.JobsRange,
+			Children:       workflowJobsSymbols(workflow),
+			Kind:           protocol.SymbolKind(ListSymbol),
+		})
+	}
+
+	if workflow.HasTrigger {
+		fmt.Println("Triggers symbols", workflow.Triggers)
```
Comment (rlegan) [inline pkg/services/documentSymbols/workflows.go:48]: to delete

File: pkg/services/documentSymbols/orbs.go [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+			},
+		)
+	}
+
+	symbol.Children = children
+
+	return []protocol.DocumentSymbol{symbol}
+}
+
+// TODO: Remove this
```
Comment (rlegan) [inline pkg/services/documentSymbols/orbs.go:40]: to delete ?

**Review by rlegan — APPROVED** [review]

**Pull request merged**